packages/flutter_driver/lib/src/driver/timeline_summary.dart

```diff
@@ -150,15 +150,19 @@ class TimelineSummary {
     Duration extractor(TimelineEvent beginEvent, TimelineEvent endEvent),
   ) {
     final List<Duration> result = <Duration>[];
+    final List<TimelineEvent> events = _extractNamedEvents(name);
```

Comment:
While the code below fixes the case with rogue events that don't come in pairs, it doesn't seem to fix the case where `_extractNamedEvents` didn't return events in time order? Maybe we need to add a `sort` somewhere, and probably assert that events are in order in the `for` loop below.

Reply:
I think it should be the responsibility of the timeline exporter to sort the events prior to producing the json meant for consumption by `chrome://tracing`. I reached out to the team, will respond here with what I hear. I think this is a good change independent of that.

> typo "not" should be "now"

Yes, it's now 👍

Reply:
I see. Since it's their responsibility to sort, we only need to do some `assert` check to make sure that they fulfill the responsibility :)

Reply:
Given that this assert could currently fail. I've filed https://github.com/flutter/flutter/issues/54300 to enable this once the observatory team has enforced this invariant.